core/src/main/java/org/apache/iceberg/io/ResolvingFileIO.java

```diff
@@ -83,21 +90,22 @@ public Map<String, String> properties() {
 
   @Override
   public void initialize(Map<String, String> newProperties) {
-    close(); // close and discard any existing FileIO instances
```

Comment:
@rdblue do you remember why this was necessary? I removed this to have a flag for detecting whether `close()` was already called

Reply:
It was just in case `initialize` was called again. Technically, that invalidates the configuration and we should close/reopen any FileIO instances that are held.

We could just reset `isClosed` in this method instead of removing this.

Reply:
I've reset `isClosed` here for now but I think we should consider removing this `close()` call from `initialize(..)` in a follow-up